File: Data Structures & Algorithms/max-water-container/submission-1.js
class Solution {
    /**
     * @param {number[]} heights
     * @return {number}
     */
    maxArea(h){
        const n=h.length;
        let i=0,j=n-1,ans=0;
        while(i<j){
            if(h[i]>h[j]){
                ans=Math.max(ans,(j-i)*h[j]);
                j--;
            }else if(h[i]<h[j]){
                ans=Math.max(ans,(j-i)*h[i]);
                i++;
            }else{
                ans=Math.max(ans,(j-i)*h[i]);
                i++;
                j--;
            }
        }
        return ans;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto-with-cooldown/submission-1.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n=prices.size();
        vector<int> dp(n,0);
        int ans=0;
        for(int i=1;i<n;i++){
            for(int j=0;j<i;j++){
                if(prices[i]>prices[j]){
                    int k=prices[i]-prices[j];
                    if(j>=2) k+=dp[j-2];
                    dp[i]=max(dp[i],k);
                }
            }
            ans=max(ans,dp[i]);
        }
        return ans;
    }
};
